src/libraries/Microsoft.Extensions.Primitives/src/StringSegmentComparer.cs

```diff
@@ -33,6 +42,11 @@ public bool Equals(StringSegment x, StringSegment y)
             return StringSegment.Equals(x, y, Comparison);
         }
 
+        /// <summary>
+        /// Returns a hash code for a <see cref="StringSegment"/>.
+        /// </summary>
+        /// <param name="obj">The <see cref="StringSegment"/> to get a hash code from.</param>
```

Comment:
`from` -> `for`?  https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.gethashcode?view=net-7.0#parameters